### PR TITLE
Adding trigger calls to non-initial login calls (issue #371)

### PR DESCRIPTION
**Pull request by billybennett**
Answered the original critiques.  Pulled against dev, indentation fixed, sloppy args, and i18n calls delayed until display.

### CODE REVIEW

File: src/view/observer.js [hunk 1]
```diff
@@ -313,10 +313,13 @@ Candy.View.Observer = (function(self, $) {
...
 	 * Parameters:
 	 *   (jQuery.Event) event - jQuery Event object
-	 *   (Object) args - {presetJid}
+	 *   (Object) args - {messageKey, presetJid}
 	 */
 	self.Login = function(event, args) {
-		Candy.View.Pane.Chat.Modal.showLoginForm(null, args.presetJid);
+		args = args || {};
+		args.messageKey = args.messageKey || '';
+		args.presetJid = args.presetJid || null;
```
Comment (benlangfeld) [inline src/view/observer.js]: I'm pretty sure you don't need these default assignments. Do them in the method you're calling and go ahead and pass undefined. This is messy :/

Comment (billybennett) [inline src/view/observer.js]: If args is undefined, args.XXX will cause an error. (XXX is anything, presetJid or messageKey in this case).

If I wanted to be insulted, I'd get married.  Use it or don't, I'm done.  Good luck with the code.

Comment (benlangfeld) [inline src/view/observer.js]: Insulted? This is code review, not a personal discussion. Why the animosity?

```javascript
var foo = {};
> undefined
foo.bar
> undefined
```

Comment (mweibel) [inline src/view/observer.js]: @billybennett we do code reviews to review code. If we don't agree with something, we tell it to you. Don't take that personally.

I agree with @benlangfeld that the default assignments are not necessary. If they don't get passed, they're `either` null or `undefined`.

File: src/view/pane/chat.js [hunk 2]
```diff
@@ -550,10 +550,11 @@ Candy.View.Pane = (function(self, $) {
        * Show the login form modal
        *
        * Parameters:
-       *  (String) message - optional message to display above the form
+       *  (String) messageKey - optional message to display above the form
        *  (String) presetJid - optional user jid. if set, the user will only be prompted for password.
        */
-      showLoginForm: function(message, presetJid) {
+      showLoginForm: function(messageKey, presetJid) {
+        var message = $.i18n._(messageKey);
```
Comment (mweibel) [inline src/view/pane/chat.js]: here's the place to do: 

```javascript
var message = messageKey ? $.i18n._(messageKey) : '';
```
Then you can remove the ternary operator for `message` on the line with `self.Chat.Modal.show` and you won't need the default assignments from above.

Comment (billybennett) [inline src/view/pane/chat.js]: Sounds like you have a good handle on it then.  I was referring to the first assignment (args = args || {}) as being necessary, if args is undefined args.presetJid will error.

Thanks for the professional response.  I haven't had my code called 'messy' in decades.  Frankly it hurt.  I tried to be sociable earlier (regarding tabs), he shot that down too.  I'm not getting paid, so it had better be pleasant; if neither applies, I have other things i can do.  But I do wish you well with the project.

Comment (melissanoelle) [inline src/view/pane/chat.js]: I'm sure I'm not alone when I say that we appreciate your drive to contribute to this project. It's clear we all care about this project. However, that also means that we want to make sure that the code that goes in here is as clean and neat as possible. We're not necessarily getting paid either. I'm not sure anything unpleasant happened here other than you became defensive of your code that you wrote and weren't open/able to receive some friendly constructive criticism. 

I hope you can consider this a growing experience and that you contribute to Candy and/or OSS in the future. :+1:

**Comment by mweibel** [comment]
:+1: to what @melissanoelle said. Sad that you removed all code now. 
Locking this comment section now.

**Pull request closed without merging**